### PR TITLE
Documentation for TOF10120 support

### PR DESCRIPTION
**Pull request by wstrzalka**
## Description:

Documentation for TOF10120 distance sensor

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/797

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1375

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/tof10120.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+Configuration variables:
+------------------------
+
+- **name** (**Required**, string): The name of the sensor.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``60s``.
+- All other options from :ref:`Sensor <config-sensor>`.
+- **address** (*Optional*, int): Manually specify the I^2C address of the sensor. Defaults to ``0x52``.
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
```
Comment (glmnet) [inline components/sensor/tof10120.rst]:
```suggestion
- **address** (*Optional*, int): Manually specify the I^2C address of the sensor. Defaults to ``0x52``.
- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
- All other options from :ref:`Sensor <config-sensor>`.
```

File: components/sensor/tof10120.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
+TOF10120 Time Of Flight Distance Sensor
+=======================================
+
+.. seo::
+    :description: Instructions for setting up TOF10120 distance sensors in ESPHome.
+    :image: vl53l0x.jpg
+    :keywords: VL53L0X
```
Comment (glmnet) [inline components/sensor/tof10120.rst]: this SEO will show the wrong sensor image

**Review by glmnet — CHANGES_REQUESTED** [review]
It will be also nice if you can use https://tinyjpg.com/ to optimize images, then it will be ok. Thanks!

File: components/sensor/tof10120.rst (hunk 2 above)
Comment (wstrzalka) [reply components/sensor/tof10120.rst]: 🤦

File: components/sensor/tof10120.rst (hunk 1 above)
Comment (wstrzalka) [reply components/sensor/tof10120.rst]: fixed

File: components/sensor/tof10120.rst (hunk 2 above)
Comment (wstrzalka) [reply components/sensor/tof10120.rst]: fixed

**Comment by wstrzalka** [comment]
> It will be also nice if you can use https://tinyjpg.com/ to optimize images, then it will be ok. Thanks!

fixed

**Review by jesserockz — APPROVED** [review]

**Pull request merged**